api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsEligibleForVLCohortQueries.java

```diff
@@ -0,0 +1,1170 @@
+            + " GROUP BY p.patient_id "
+            + "                      "
+            + " UNION "
+            + "  "
+            + " SELECT p.patient_id, most_recent.last_obs AS recent_date FROM patient p "
+            + " INNER JOIN encounter e ON e.patient_id = p.patient_id "
+            + " INNER JOIN obs o ON o.encounter_id = e.encounter_id "
+            + " INNER JOIN "
+            + " (SELECT p.patient_id, MAX(o.value_datetime) AS last_obs, o.obs_datetime "
+            + " FROM encounter e INNER JOIN obs o ON o.encounter_id = e.encounter_id "
+            + " INNER JOIN patient p ON p.patient_id = e.patient_id "
+            + " WHERE e.encounter_type = ${52} "
+            + " AND o.concept_id = ${23866} "
+            + "          "
+            + " AND e.voided = 0 "
+            + " AND e.location_id = :location "
+            + "         AND e.voided = 0  "
+            + "         AND p.voided = 0 "
+            + "         AND o.voided = 0 "
+            + " GROUP BY p.patient_id) most_recent "
+            + " ON most_recent.patient_id = e.patient_id "
+            + " WHERE p.voided = 0 "
+            + "   AND e.encounter_type = ${52} "
+            + "   AND e.location_id = :location "
+            + "   AND o.concept_id = ${23865} "
+            + "   AND o.value_coded = ${1065} "
+            + "   AND o.obs_datetime = most_recent.obs_datetime "
+            + "   AND o.voided = 0 "
+            + "   AND e.voided = 0 "
+            + "   AND p.voided = 0 "
+            + "   AND DATE_ADD(most_recent.last_obs, INTERVAL 30 DAY) BETWEEN :startDate AND :endDate "
+            + "   AND most_recent.last_obs <= :endDate "
+            + " GROUP BY p.patient_id) AS most_recentx "
+            + " GROUP BY most_recentx.patient_id "
+            + " ) AS xpatient "
+            + " ) x ON x.patient_id = most_recentvl.patient_id "
+            + " AND TIMESTAMPDIFF(MONTH,  most_recentvl.recent_date,x.recentx_date) >= 12 "
+            + " GROUP BY x.patient_id ";
+
+    StringSubstitutor stringSubstitutor = new StringSubstitutor(valuesMap);
+
+    sqlCohortDefinition.setQuery(stringSubstitutor.replace(query));
+
+    return sqlCohortDefinition;
+  }
+
+  /**
+   * <b>Technical Specs</b>
+   *
+   * <blockquote>
+   *
+   * <p>VL5- select all patients with the most recent VL Numeric Result (concept Id 856) documented
+   * in the Laboratory Form (encounter type 13, encounter_datetime) or Ficha de Seguimento Adulto or
+   * Pediatria (encounter type 6,9, encounter_datetime) or Ficha Clinica (encounter type 6,
+   * encounter_datetime) or Ficha Resumo (encounter type 53, obs_datetime ) or FSR form (encounter
+   * type 51, encounter_datetime) by start of reporting period (<=startDate) and the Result is >=
+   * 1000 copias/ml (concept 856 value_numeric >= 1000)
+   *
+   * </blockquote>
+   *
+   * @return {@link CohortDefinition}
+   */
+  public CohortDefinition getPatientsWithRecentVLIgualOrGreaterThan1000() {
```

Comment:
@eliazmuchave can you update this to be generic as the name of this function getPatientsWithVLLessThan1000()

Reply:
I'm not clear